bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/handler/CloudBridgeHandler.java

```diff
@@ -0,0 +1,263 @@
+                .build(this::loadPropertiesForDevice);
+        var scheduledPool = ThreadPoolManager.getScheduledPool("Salus");
+        this.scheduledFuture = scheduledPool.scheduleWithFixedDelay(this::refreshCloudDevices,
+                config.getRefreshInterval() * 2, config.getRefreshInterval(), SECONDS);
+
+        // done
+        updateStatus(ONLINE);
+    }
+
+    private void refreshCloudDevices() {
+        logger.debug("Refreshing devices from CloudBridgeHandler");
+
+        if (!(thing instanceof Bridge bridge)) {
+            logger.debug("No bridge, refresh cancelled");
+            return;
+        }
+        var things = bridge.getThings();
+        for (var thing : things) {
+            if (!thing.isEnabled()) {
+                logger.debug("Thing {} is disabled, refresh cancelled", thing.getUID());
+                continue;
+            }
+            var handler = thing.getHandler();
+            if (handler == null) {
+                logger.debug("No handler for thing {} refresh cancelled", thing.getUID());
+                continue;
+            }
+            var channels = thing.getChannels();
+            for (var channel : channels) {
+                handler.handleCommand(channel.getUID(), REFRESH);
+            }
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // no commands in this bridge
+        logger.debug("Bridge does not support any commands to any channels. channelUID={}, command={}", channelUID,
+                command);
+    }
+
+    @Override
+    public void dispose() {
+        var localScheduledFuture = scheduledFuture;
+        if (localScheduledFuture != null) {
+            localScheduledFuture.cancel(true);
+            scheduledFuture = null;
+        }
+        super.dispose();
+    }
+
+    @Override
+    public SortedSet<@NonNull DeviceProperty<?>> findPropertiesForDevice(String dsn) {
+        return requireNonNullElse(devicePropertiesCache.get(dsn), emptySortedSet());
+    }
+
+    @Nullable
+    private SortedSet<DeviceProperty<?>> loadPropertiesForDevice(String dsn) {
+        var api = salusApi;
+        if (api == null) {
+            logger.error("Cannot find properties for device {} because salusClient is null", dsn);
+            return null;
+        }
+        logger.debug("Finding properties for device {} using salusClient", dsn);
+        var response = api.findDeviceProperties(dsn);
+        if (response.failed()) {
+            logger.error("Cannot find properties for device {} using salusClient\n{}", dsn, response.error());
+            return null;
+        }
+        return response.body();
+    }
+
+    @Override
+    public boolean setValueForProperty(String dsn, String propertyName, Object value) {
+        var api = salusApi;
+        if (api == null) {
+            logger.error("Cannot set value for property {} on device {} because salusClient is null", propertyName,
+                    dsn);
```

Comment:
logger.debug("Salus client is null and that should never happen");

Reply:
This is kind of problematic place. At this point `salusApi` should be set (in `initilization` method`), but this property is set as nullable (for properties you need to set it as nullable or init it in constructor). 

What's your advice?

Reply:
ping